src/Language/Haskell/Liquid/Types/Errors.hs

```diff
@@ -1043,6 +1049,10 @@ ppError' _ dCtx (ErrRewrite _ msg )
         $+$ dCtx
         $+$ nest 4 msg
 
+ppError' _ dCtx (ErrPosTyCon _ tc dc)
+  = text "Non strictly positive occurence of" <+> tc <+> "in" <+> dc 
+        $+$ dCtx
```

Comment:
If the error could reference a github issue or some other pointer discussing the purpose of the check the user might have a better time figuring it out.

Also, maybe the error message could mention the flag to disable the check and warn appropriately about using it.